mne/io/base.py

```diff
@@ -2503,12 +2503,14 @@ def _check_raw_compatibility(raw):
     for ri in range(1, len(raw)):
         if not isinstance(raw[ri], type(raw[0])):
             raise ValueError(f'raw[{ri}] type must match')
-        for key in ('nchan', 'bads', 'sfreq'):
+        for key in ('nchan', 'sfreq'):
             a, b = raw[ri].info[key], raw[0].info[key]
             if a != b:
                 raise ValueError(
                     f'raw[{ri}].info[{key}] must match:\n'
                     f'{repr(a)} != {repr(b)}')
+        if not set(raw[ri].info['bads']) == set(raw[0].info['bads']):
+            raise ValueError('raw[%d][\'info\'][\'bads\'] must match' % ri)
```

Comment:
... also this line is less informative than the one above because it doesn't give the differences. You could use something like `sorted(set(a).symmetric_difference(set(b)))` to get the entries that don't match and print them like the line above